test/dataset_test.cpp

```diff
@@ -1073,3 +1073,128 @@ TEST(DatasetSlice, subset_slice_spatial_with_bin_edges) {
       view_a_x13 -= view_a_x02, std::runtime_error,
       "Coordinates of datasets do not match. Cannot perform binary operation.");
 }
+
+TEST(Dataset, unary_minus) {
+  Dataset a;
+  a.insert<Coord::X>({Dim::X, 2}, {1, 2});
+  a.insert<Data::Value>("a", {Dim::X, 2}, {1, 2});
```

Comment:
May i suggest to have something different from the name of the Datasets inside the strings, i.e. replace `"a"` and `"b"` with something like `val1` and `val2` to avoid a bit of confusion? i.e.
```
Dataset a;
a.insert<Data::Value>("val1", {Dim::X, 2}, {1, 2});
```